packages/apps/intervention/src/components/templates/SurveyRangeNavigation/SurveyRangeNavigation.tsx

```diff
@@ -29,28 +29,28 @@ const ARROW_TYPE = {
 };
 
 interface SurveyRangeNavigationProps {
-  defaultValue?: string;
+  selectedRangeId: number;
   onChangeRangeId: (rangeId: number) => void;
 }
 
 const SurveyRangeNavigation = ({
-  defaultValue = null,
-  onChangeRangeId = () => {},
+  selectedRangeId,
+  onChangeRangeId,
 }: SurveyRangeNavigationProps) => {
   const dispatch = useDispatch();
 
-  const [selectedRangeId, setSelectedRangeId] = useState(null);
-
-  useEffect(() => setSelectedRangeId(defaultValue), [defaultValue]);
-
   const {intervention} = useSelector(
     (state: any) => state.intervention_intervention,
   );
   const {rangeList} = useSelector((state: any) => state.intervention_question);
 
   useEffect(() => {
-    dispatch((fetchRange as any)({interventionId: intervention?.id}));
-  }, [dispatch, intervention?.id]);
+    (dispatch as any)(
+      (fetchRange as any)({interventionId: intervention?.id}),
+    ).then(
+      res => selectedRangeId == null && onChangeRangeId(res.payload[0].id),
+    );
+  }, [dispatch, intervention?.id, onChangeRangeId, selectedRangeId]);
```

Comment:
Move this to InterventionDetailsScreen and caution to state and state setter in the same useEffet, you can use the current state directly from the setter for your condition